### PR TITLE
feat(cloudflare): add support for cloudflare's image cdn

### PR DESCRIPTION
**Pull request by nicksrandall**
Fixes: #1

### CODE REVIEW

**Comment by nicksrandall** [comment]
I'm excited to delete the custom, 100+ line image component in my app and replace it with this -- I just need it to support Cloudflare :)

If it's not too much trouble, could you please push a new version to NPM of this and `unpic-img` when (if?) this PR is merged?

File: src/transformers/cloudflare.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import {
+  UrlGenerator,
+  UrlGeneratorOptions,
+  UrlParser,
+  UrlTransformer,
+} from "../types.ts";
+
+// Thanks Colby!
```
Comment (ascorbic) [inline src/transformers/cloudflare.ts:8]: Probably don't need to thank Colby for this one!

File: src/transformers/cloudflare.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  Object.fromEntries(transformations.split(",").map((t) => t.split("=")));
+
+const formatUrl = (
+  {
+    host,
+    transformations = {},
+    path
+  }: CloudflareParams,
+): string => {
+  console.log('trans', transformations);
```
Comment (ascorbic) [inline src/transformers/cloudflare.ts:22]: Left this in

**Review by ascorbic — CHANGES_REQUESTED** [review]
Thanks for this! Are there any default domains that it can auto-detect, or is it always the user's domain? Could you share a real example image to test it, preferably one from the Cloudflare docs.

Could you also add it to the README, but with a note saying that it's not auto-detected and needs to be manually specified. Thanks again!

**Comment by nicksrandall** [comment]
I'll get your feedback implemented. 

Regarding auto detection by domain, that is not possible with cloudflare because it always uses the users domain. That said, it would be possible to detect by a pathname prefix (`/cdn-cgi/image`). Do you think its worth adding that logic (similar to how you currently do subdomains)?

**Comment by nicksrandall** [comment]
@ascorbic I addressed your feedback. I wasn't able to find an image in the docs that works but I did change the URL to match what is in the docs. 

Also, I was able to add Cloudflare to the detection function by looking for a specific path prefix. Let me know if you think that is acceptable.

File: src/transformers/cloudflare.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { assertEquals } from "https://deno.land/std@0.172.0/testing/asserts.ts";
+import { ParsedUrl } from "../types.ts";
+import { CloudflareParams, parse, transform } from "./cloudflare.ts";
+
+const img =
+  "https://assets.example.comeight=64,f=auto,background=red/uploads/avatar1.jpg"
```
Comment (ascorbic) [inline src/transformers/cloudflare.test.ts:6]: I think something's wriong with this URL!

File: src/transformers/cloudflare.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      transformations: {
+        background: "red",
+      },
+      path: "uploads/340210587049918541/group_141-1664915283181.png",
+    },
+  };
+  assertEquals(parsed, expected);
+});
+
+Deno.test("cloudfalre transformer", async (t) => {
```
Comment (ascorbic) [inline src/transformers/cloudflare.test.ts:27]:
```suggestion
Deno.test("cloudflare transformer", async (t) => {
```

**Review by ascorbic — COMMENTED** [review]
Good idea with the path matching. I think there are some others that we could use that approach for. Just a couple of typos left. I would really like if we can find at least one valid URL that can be used to demo this, so it can be added to the playground in examples.json. Otherwise I've no way of being sure it actually works. Comparing it to the docs only goes so far.

**Comment by ascorbic** [comment]
The tests are failing because of that broken url

**Comment by nicksrandall** [comment]
Fixed the test with a generic image from my app. Hopefully that is good enough?

File: src/transformers/cloudflare.test.ts (hunk 3 above)
Comment (nicksrandall) [reply src/transformers/cloudflare.test.ts]: Fixed

**Review by ascorbic — APPROVED** [review]
Perfect! Thanks for your help

**Pull request merged**